drivers/flash/flash_spi.c

```diff
@@ -0,0 +1,382 @@
+    response= spi_flash_wait_while_busy();
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to check WIP flag! [spi flash erase sector]\n");
+        return response;
+    }
+
+    return response;
+}
+
+flash_status_t flash_erase_block(const void *addr) {
+    flash_status_t response    = FLASH_STATUS_SUCCESS;
+    uintptr_t      target_addr = (uintptr_t)addr;
+
+    /* Check that the address exceeds the limit. */
+    if ((target_addr + (EXTERNAL_FLASH_BLOCK_SIZE << 10)) >= (EXTERNAL_FLASH_SIZE << 10) ||
+        ((target_addr % (EXTERNAL_FLASH_BLOCK_SIZE << 10)) != 0)) {
+        dprintf("Flash erase block address over limit! [addr:0x%x]\n", (uint32_t)target_addr);
+        return FLASH_STATUS_ERROR;
+    }
+
+    /* Wait for the write-in-progress bit to be cleared. */
+    response= spi_flash_wait_while_busy();
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to check WIP flag! [spi flash erase block]\n");
+        return response;
+    }
+
+    /* Enable writes. */
+    response = spi_flash_write_enable();
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to write-enable! [spi flash erase block]\n");
+        return response;
+    }
+
+    /* Erase Block. */
+    response = spi_flash_transmit((uint8_t)FLASH_CMD_BE, target_addr, NULL, 0);
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to erase block! [spi flash erase block]\n");
+        return response;
+    }
+
+    /* Wait for the write-in-progress bit to be cleared.*/
+    response= spi_flash_wait_while_busy();
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to check WIP flag! [spi flash erase block]\n");
+        return response;
+    }
+
+    return response;
+}
+
+flash_status_t flash_read_block(uint8_t *buf, const void *addr, size_t len) {
+    flash_status_t response    = FLASH_STATUS_SUCCESS;
+    uint8_t *      read_buf    = (uint8_t *)buf;
+    uintptr_t      target_addr = (uintptr_t)addr;
+
+    /* Wait for the write-in-progress bit to be cleared. */
+    response= spi_flash_wait_while_busy();
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to check WIP flag! [spi flash read block]\n");
+        memset(read_buf, 0, len);
+        return response;
+    }
+
+    /* Perform read. */
+    response = spi_flash_transmit((uint8_t)FLASH_CMD_READ, (uintptr_t)target_addr, read_buf, len);
+    if (response != FLASH_STATUS_SUCCESS) {
+        dprint("Failed to read block! [spi flash read block]\n");
+        memset(read_buf, 0, len);
+        return response;
+    }
+
+#if defined(CONSOLE_ENABLE) && defined(DEBUG_FLASH_SPI_OUTPUT)
+    dprintf("[SPI FLASH R] 0x%08lX: ", ((uint32_t)target_addr));
+    for (size_t i = 0; i < len; ++i) {
+        dprintf(" %02X", (int)(((uint8_t *)read_buf)[i]));
+    }
+    dprintf("\n");
+#endif  // DEBUG_FLASH_SPI_OUTPUT
+
+    return response;
+}
+
+flash_status_t flash_write_block(const uint8_t *buf, void *addr, size_t len) {
```

Comment:
```suggestion
flash_status_t flash_write_block(uint32_t addr, const void *buf, size_t len) {
```
As per previous, swap over usages of `target_addr`.